src/languages/scala.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  createPatternMatchers,
+  argumentMatcher,
+  leadingMatcher,
+  conditionMatcher,
+  trailingMatcher,
+  cascadingMatcher,
+} from '../util/nodeMatchers';
+import { NodeMatcherAlternative, ScopeType } from '../typings/Types';
+
+const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
+  // treating classes = classlike
+  class: ['class_definition', 'object_definition', 'trait_definition'],
+  className: ['class_definition[name]', 'object_definition[name]', 'trait_definition[name]'],
+
+  ifStatement: 'if_expression',
+
+  string: ['interpolated_string_expression', 'string'],
+  comment: 'comment',
+
+  // list.size(), does not count foo.size (field_expression), or foo size (postfix_expression)
+  functionCall: 'call_expression',
+  namedFunction: 'function_definition',
+  anonymousFunction: 'lambda_expression',
+
+  argumentOrParameter: argumentMatcher('arguments', 'parameters', 'class_parameters', 'bindings'),
+
+  name: ['*[name]', '*[pattern]'],
```

Comment:
interesting.  we don't do this in typescript, but maybe we should 🤔 cc/ @AndreasArvidsson

Reply:
What is it we don't do in typescript? The current implementation looks like this
```ts
name: [
    "*[name]",
    "optional_parameter.identifier!",
    "required_parameter.identifier!",
  ]
```

Reply:
Oh yeah just tested and looks like it works.  The example was

```typescript
const {foo, bar} = baz;
```

Looks like saying "take name" already selects `{foo, bar}`, which is what I was proposing, so 👍